src/test/java/com/microsoft/azure/cosmosdb/kafka/connect/source/integration/SourceConnectorIT.java

```diff
@@ -156,44 +167,130 @@ private Properties createKafkaConsumerProperties() {
         return kafkaProperties;
     }
 
+    private Optional<ConsumerRecord<String, JsonNode>> searchConsumerRecords(Person person) {
+        ConsumerRecords<String, JsonNode> records = consumer.poll(Duration.ofMillis(2000));
+        for (ConsumerRecord<String, JsonNode> record : records) {
+            recordBuffer.add(record);
+        }
+
+        return recordBuffer.stream().filter(
+            p -> p.value().get("id").textValue().equals(person.getId())).findFirst();
+    }
+
     /**
-     * Create an item in Cosmos DB and have the source connector transfer data into a Kafka topic.
-     * Then read the result from Kafka topic.
+     * Create a source connector and an item in Cosmos DB. Allow the source connector to 
+     * transfer data into a Kafka topic. Then read the result from Kafka topic.
      */
     @Test
     public void testReadCosmosItem() throws InterruptedException, ExecutionException {
-        // Configure Kafka Config
-        Properties kafkaProperties = createKafkaConsumerProperties();
-
         // Create source connector with default config
         connectClient.addConnector(connectConfig.build());
+        
+        // Allow time for Source connector to setup resources
+        sleep(8000);
+
+        // Create item in Cosmos DB
+        Person person = new Person("Lucy Ferr", RandomUtils.nextLong(1L, 9999999L) + "");
+        targetContainer.createItem(person);
+
+        // Allow time for Source connector to process data from Cosmos DB
+        sleep(8000);
+
+        Optional<ConsumerRecord<String, JsonNode>> resultRecord = searchConsumerRecords(person);
+        Assert.assertNotNull("Person could not be retrieved from messages", resultRecord.orElse(null));
+        Assert.assertTrue("Message Key is not the Person ID", resultRecord.get().key().contains(person.getId()));
+    }
+
+    /**
+     * Testing the connector's response to offsets.
```

Comment:
nice test. 
we should test this with the following - 
a) multiple workers
b) multiple cosmos db partitions

a single worker should process a single partition (to preserve order info). if a worker dies, another one should be able to resume from the same point in the partition the one that died got to, or start processing a new partition.